internal/pkg/agent/application/application.go

```diff
@@ -111,7 +113,7 @@ func New(
 		}
 	}
 
-	composable, err := composable.New(log, rawConfig)
```

Comment:
seems like we should address changes from #848 and #686  here as well

Reply:
yeah hopefully I got all that in the merge, I ensured all the reload interfaces are present